### DISCUSSION TITLE
How write logs in MinIO (\w Laravel) ??

### DISCUSSION THREAD

**Discussion by a-h-abid** [minio/minio#13616] [Q&A]
Hi there,

I'm using MinIO with Laravel (PHP) to store uploaded data, it is working great.

Now I have a requirement to store logs in MinIO. I searched but could not find a solution or package. 

Can anyone tell me how can I do it ??

**Edit:** Just to be clear I meant storing application generated logs in MinIO (in my case Laravel logs).

**Comment by klauspost** [comment]
https://docs.min.io/minio/baremetal/monitoring/logging/minio-logging.html?#logging

**Reply by a-h-abid** [reply]
> https://docs.min.io/minio/baremetal/monitoring/logging/minio-logging.html?#logging

Thanks but I'm looking for storing application logs (in my case Laravel logs).

Sorry I wasn't clear before.

**Reply by klauspost** [reply]
@a-h-abid I am still not sure I understand your question. MinIO (or any S3 service) doesn't really care what you store in it. You upload logs just like you upload any other data.

There is no streaming/append functionality in S3 if that is what you are looking for. So you will probably need to store them locally and then rotate logs to S3 at regular intervals.

**Reply by a-h-abid** [reply]
Yes, streaming/appending to log file is what I was looking for. I guess I have to look for alternative way.

Thanks.

**Reply by bpedersen2** [reply]
I would consider one of the log aggregating  systems like LOKI (that could  be backed by S3 compat storage.)
